src/main/kotlin/org/jitsi/videobridge/ice/IceConfig.kt

```diff
@@ -0,0 +1,187 @@
+             */
+            class TcpPortProperty : LegacyFallbackConfigProperty<Int>(
+                Int::class,
+                readOnce = true,
+                legacyName = "org.jitsi.videobridge.TCP_HARVESTER_PORT",
+                newName = "videobridge.ice.tcp.port"
+            )
+            private val tcpPortProperty = TcpPortProperty()
+
+            /**
+             * The property which configures the fallback ICE/TCP port.
+             */
+            class TcpFallbackPortProperty : SimpleProperty<Int>(
+                attributes {
+                    name("videobridge.ice.tcp.fallback-port")
+                    readOnce()
+                    fromConfig(JitsiConfig.newConfig)
+                }
+            )
+            private val tcpFallbackPortProperty = TcpFallbackPortProperty()
+
+            /**
+             * The list of ports to try. Fallback can be disabled by setting the fallback port to <=0.
+             */
+            @JvmStatic
+            fun tcpPortsToTry() = when {
+                (tcpFallbackPortProperty.value > 0) ->
+                    listOf(tcpPortProperty.value, tcpFallbackPortProperty.value)
+                else -> listOf(tcpPortProperty.value)
+            }
+
+            /**
+             * The property which configures an additional port to advertise.
+             */
+            class TcpMappedPortProperty : LegacyFallbackConfigProperty<Int>(
+                Int::class,
+                readOnce = true,
+                legacyName = "org.jitsi.videobridge.TCP_HARVESTER_MAPPED_PORT",
+                newName = "videobridge.ice.tcp.mapped-port"
+            )
+            private val tcpMappedPortProperty = TcpMappedPortProperty()
+
+            /**
+             * Returns the additional port to advertise, or [null] if none is configured.
+             */
+            @JvmStatic
+            fun tcpMappedPort(): Int? = try {
+                tcpMappedPortProperty.value
+            } catch (e: Throwable) {
+                null
+            }
+
+            /**
+             * The property that configures whether ICE/TCP should use "ssltcp" or not.
+             */
+            class SslTcpProperty : LegacyFallbackConfigProperty<Boolean>(
+                Boolean::class,
+                readOnce = true,
+                legacyName = "org.jitsi.videobridge.TCP_HARVESTER_SSLTCP",
+                newName = "videobridge.ice.tcp.ssltcp"
+            )
+            private val sslTcpProperty = SslTcpProperty()
+
+            @JvmStatic
+            fun iceSslTcp() = sslTcpProperty.value
+
+            /**
+             * The property that configures the ICE UDP port.
+             */
+            class PortProperty : LegacyFallbackConfigProperty<Int>(
+                Int::class,
+                readOnce = true,
+                legacyName = "org.jitsi.videobridge.SINGLE_PORT_HARVESTER_PORT",
+                newName = "videobridge.ice.port"
```

Comment:
Think this needs to change as well to match the new location `videobridge.ice.udp.port`